### PR TITLE
Ensure updating `Context.Consumer` inside suspended `Suspense` component

### PR DESCRIPTION
**Pull request by okmttdhr**
## Summary

Ensure updating `Context.Consumer` inside suspended `Suspense` component when context updates. Please see https://github.com/facebook/react/issues/19701 for details of this bug.

Fixes https://github.com/facebook/react/issues/19701

Related https://github.com/facebook/react/pull/14717 https://github.com/facebook/react/pull/16346 | https://github.com/facebook/react/pull/20890 | https://github.com/facebook/react/pull/18796

## Test Plan

Run `yarn test --watch ReactSuspense-test.internal`.

### CODE REVIEW

File: packages/react-reconciler/src/ReactFiberNewContext.new.js [hunk 1]
```diff
@@ -157,10 +157,6 @@ export function scheduleWorkOnParentPath(
       !isSubsetOfLanes(alternate.childLanes, renderLanes)
     ) {
       alternate.childLanes = mergeLanes(alternate.childLanes, renderLanes);
-    } else {
-      // Neither alternate was updated, which means the rest of the
-      // ancestor path already has sufficient priority.
```
Comment (okmttdhr) [inline packages/react-reconciler/src/ReactFiberNewContext.new.js:162]: This comment may not be correct in cases where `Context.Consumer` is the ancestor of `Suspense`.

(But with this PR, it goes to all the ancestors, so please let me know if you have a better way to do it)

Comment (gaearon) [reply packages/react-reconciler/src/ReactFiberNewContext.new.js:162]: This does seem pretty aggressive, right? It seems like this is an important optimization here. I think you need to figure out why the assumption in this code is not true, and how we can either make it true, or change the check in a way that doesn't regress all other cases.

Comment (okmttdhr) [reply packages/react-reconciler/src/ReactFiberNewContext.new.js:162]: Thank you for your comment, I'll figure the root cause out.

※Current situation; it [bailoutOnAlreadyFinishedWork](https://github.com/okmttdhr/react/blob/fb36d983ee5684ab0994eccc7d670a6a822ae35d/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L3337) on a function component that wraps `Context.Consumer` before it get to `Context.Consumer` ([This childLanes](https://github.com/okmttdhr/react/blob/fb36d983ee5684ab0994eccc7d670a6a822ae35d/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L3113) would be `Context.Consumer`'s parent)

Comment (okmttdhr) [reply packages/react-reconciler/src/ReactFiberNewContext.new.js]: @gaearon 
Changed it to schedule only needed work.
https://github.com/facebook/react/pull/21414/commits/ed589927ba84cb36342918b26f7062be3d927ef9

**Comment by okmttdhr** [comment]
bump

**Comment by gaearon** [comment]
Another possible fix: https://github.com/facebook/react/pull/23095. Not sure what's better.

**Comment by okmttdhr** [comment]
> Another possible fix: #23095. Not sure what's better.

Let me briefly summarize for my understanding.

First of all, this PR is an approach to "directly" update `nodes` that can be inconsistent only when it knows that there is `Consumer` in `SuspenseComponent`. There is a possibility of traverse outside of the propagation root, but if it is consistent, it will `break;` anyway, so I don't see a problem.

But this PR only fixes the case where the bug in #19701 occurs.
For example, as you mentions in #23095, it probably won't fix the cases other than #19701 where context fails to propagate (not sure if that exists).

I understood #23095 to be an approach that always traverse up to the propagation root, even if it looks consistent.
From the model perspective, "traverse upwards (past the propagation root) should not be necessary to do", so #23095 update the `childLanes` only up to the propagation root.
However, I understood that you are worrying about the case where the assumption itself is wrong, as it says;

>My concern isn't so much Consumer API but the flaw in modeling. The assumption seems fundamentally flawed and I'm worried how else it might bite us.

**Comment by gaearon** [comment]
We went with https://github.com/facebook/react/pull/23095.

**Pull request closed without merging**